AST: re-order copy constructor, remove extra copy constructor

compnerd opened the pull request:
Visual Studio objects with the following:

  ```
  AST/AnyRequest.h(243): error: C2580: 'swift::AnyRequest::AnyRequest(const swift::AnyRequest &)': multiple versions of a defaulted special member functions are not allowed
  AST/AnyRequest(243): warning C4521: 'swift::AnyRequest': multiple copy constructors specified
  ```

Remove the non-const implementation, the default'ed definition should be
sufficient.  Additionally, re-order to keep the const and non-const r-value ref
versions next to each other.

<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

compnerd commented:
CC: @DougGregor

jckarter commented on include/swift/AST/AnyRequest.h:
```diff
@@ -150,6 +150,9 @@ class AnyRequest {
 
+  AnyRequest(const AnyRequest &&other)
+    : storageKind(other.storageKind), stored(other.stored) { }
+
```
A `const &&` move constructor is useless, and probably a typo. It looks like it can be removed and left as default as well.

compnerd replied:
@jckarter yeah, that's what I thought too ... and then realized that they actually made it do something that doesn't make sense *shrug*.  I'll happily remove it.

compnerd committed:
AST: remove extra copy constructors

Visual Studio objects with the following:

```
  AST/AnyRequest.h(243): error: C2580: 'swift::AnyRequest::AnyRequest(const swift::AnyRequest &)': multiple versions of a defaulted special member functions are not allowed
  AST/AnyRequest(243): warning C4521: 'swift::AnyRequest': multiple copy constructors specified
```

Remove the non-const implementation, the default'ed definition should be
sufficient.  Additionally, remove the nonsense const move constructor.

compnerd commented:
@swift-ci please test and merge

compnerd commented:
@swift-ci please test and merge
